### DISCUSSION TITLE
Typescript - i can't change onPaste()'s behaviors

### DISCUSSION THREAD

**Discussion by pipidelena** [codex-team/editor.js#2850] [Q&A]
Hi.
https://editorjs.io/paste-substitutions - I followed instructions from the Editor.js documentation on paste substitutions. My files now look like these:
index.d.ts
![index-d-ts](https://github.com/user-attachments/assets/9ab126b4-89a3-444f-89bb-085f82ec6cf3)
configuration.ts
![configuration-ts](https://github.com/user-attachments/assets/9c54db9d-bc0b-4823-8d1b-258bbb91057f)
init function
![init-editor](https://github.com/user-attachments/assets/027f4f35-1b83-4922-b8d1-8a4201a25b5e)
testing result
![rendered](https://github.com/user-attachments/assets/f37bf440-6cc3-42b4-acc3-a4c08b137157)

As seen in the images @. @ i don't think the render() function in either CustomParagraph or Paragraph gets called. But if so what did? and how could I remove HTML tags like -a- or <script> from the pasted content in my project?  I’ve spent a whole day on this issue without any clues, please help!
* In my case, I think that modifying onPaste() is my only option since I still need to add -a- tags later in the posts and sanitizing doesn't seem to fit.

**Comment by pipidelena** [comment]
so how does the inlineToolbar work? Specifically, how is pasted data treated during the paste action when we set inlineToolbar: true, which lead to the failure of tags removal? Could someone pls provide documentation or instructions > . >

**Comment by virgo-fondasys** [comment]
someone help to answer this please, i need the answer too. @gohabereg  @neSpecc
